app/context/authContext.tsx

```diff
@@ -0,0 +1,175 @@
+'use client';
+
+import React, { createContext, useContext, ReactNode, useMemo } from 'react';
+import { MutationStatus, QueryStatus, useMutation, useQuery } from '@tanstack/react-query';
+import { clearSession, getSession, setSession } from '../../utils/actions/session';
+
+const baseUrl = process.env.NEXT_PUBLIC_API_BASE_URL || '';
+
+export type Role = 'admin' | 'content_manager' | 'member';
+
+type UserType = {
+  id: number | string;
+  is_verified: boolean;
+  email: string;
+  role: Array<Role>;
```

Comment:
This was an array initially but we changed it to be a string for now. Only one role per user will be used for the time being. It may be changed in the near future if requirements change.

Reply:
@stefanatanasovmk

Reply:
we should add role automatically, depending from where the user logs in

Reply:
@davorminchorov This is a `UserType`, I'll change the role to be of type `Role` and not an array.
@edichoska The user role is fetched from the backend. When login in, the `login` function expect an argument `role` because the backend have different routes for login depending of the user role. But, I assume that the user role will be assigned on registration on the backend, as there is different urls for login, I guess there will be different urls for registration depending of the user's role.